catalog/plugin/catalog-plugin-security-audit/src/main/java/org/codice/alliance/catalog/plugin/security/audit/SecurityAuditPlugin.java

```diff
@@ -0,0 +1,134 @@
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package org.codice.alliance.catalog.plugin.security.audit;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.collections.MapUtils;
+import org.apache.shiro.util.StringUtils;
+
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.Request;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.plugin.AccessPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.security.common.audit.SecurityLogger;
+
+public class SecurityAuditPlugin implements AccessPlugin {
+
+    private List<String> auditAttributes = new ArrayList<>();
+
+    @Override
+    public CreateRequest processPreCreate(CreateRequest createRequest)
+            throws StopProcessingException {
+        return createRequest;
+    }
+
+    @Override
+    public UpdateRequest processPreUpdate(UpdateRequest updateRequest,
+            Map<String, Metacard> existingMetacards) throws StopProcessingException {
+        if (updateRequest == null || !isLocal(updateRequest)
+                || MapUtils.isEmpty(existingMetacards)) {
+            return updateRequest;
+        }
+        for (Map.Entry<Serializable, Metacard> entry : updateRequest.getUpdates()) {
+            Metacard updateMetacard = entry.getValue();
+            Metacard existingMetacard = existingMetacards.get(entry.getKey()
+                    .toString());
+            Set<AttributeDescriptor> attributeDescriptors = updateMetacard.getMetacardType()
+                    .getAttributeDescriptors();
+            for (AttributeDescriptor descriptor : attributeDescriptors) {
+                String descriptorName = descriptor.getName();
+                if (auditAttributes.contains(descriptorName) && existingMetacard.getAttribute(
+                        descriptorName) != null) {
+                    Serializable newValue;
+                    if (updateMetacard.getAttribute(descriptorName) != null) {
+                        newValue = updateMetacard.getAttribute(descriptorName)
+                                .getValue();
+                    } else {
+                        newValue = StringUtils.EMPTY_STRING;
```

Comment:
Yeah, looking back on this, I am kind of confused myself. I think what I was trying to achieve here is the case when the attribute has been deleted.